homeassistant/components/fibaro/sensor.py

```diff
@@ -95,56 +96,34 @@ async def async_setup_entry(
 class FibaroSensor(FibaroDevice, SensorEntity):
     """Representation of a Fibaro Sensor."""
 
-    def __init__(self, fibaro_device):
+    def __init__(self, fibaro_device: Any) -> None:
         """Initialize the sensor."""
-        self.current_value = None
-        self.last_changed_time = None
         super().__init__(fibaro_device)
         self.entity_id = ENTITY_ID_FORMAT.format(self.ha_id)
+
+        self._attr_native_unit_of_measurement = None
         if fibaro_device.type in SENSOR_TYPES:
-            self._unit = SENSOR_TYPES[fibaro_device.type][1]
-            self._icon = SENSOR_TYPES[fibaro_device.type][2]
-            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
+            self._attr_native_unit_of_measurement = SENSOR_TYPES[fibaro_device.type][1]
+            self._attr_icon = SENSOR_TYPES[fibaro_device.type][2]
+            self._attr_device_class = SENSOR_TYPES[fibaro_device.type][3]
             self._attr_state_class = SENSOR_TYPES[fibaro_device.type][4]
-        else:
-            self._unit = None
-            self._icon = None
-            self._device_class = None
+
         with suppress(KeyError, ValueError):
-            if not self._unit:
-                if self.fibaro_device.properties.unit == "lux":
-                    self._unit = LIGHT_LUX
-                elif self.fibaro_device.properties.unit == "C":
-                    self._unit = TEMP_CELSIUS
-                elif self.fibaro_device.properties.unit == "F":
-                    self._unit = TEMP_FAHRENHEIT
+            if not self._attr_native_unit_of_measurement:
+                unit = self.fibaro_device.properties.unit
+                if unit == "lux":
+                    self._attr_native_unit_of_measurement = LIGHT_LUX
+                elif unit == "C":
+                    self._attr_native_unit_of_measurement = TEMP_CELSIUS
+                elif unit == "F":
+                    self._attr_native_unit_of_measurement = TEMP_FAHRENHEIT
```

Comment:
This could be a `dict` of mapping

self._attr_native_unit_of_measurement = FIBARO_TO_HASS_UNIT.get(unit)

Reply:
Makes absolutely sense, I changes that.